pywinauto/UIAElementInfo.py

```diff
@@ -167,29 +167,46 @@ def parent(self):
         else:
             return None
 
-    @property
-    def children(self):
-        "Return list of immediate children for the element"
-        children = []
-        
-        childrenArray = self._element.FindAll(_tree_scope['children'], _true_condition)
-        for childNumber in range(childrenArray.Length):
-            childElement = childrenArray.GetElement(childNumber)
-            children.append(UIAElementInfo(childElement))
+    def _get_elements(self, scope, cond=_true_condition):
+        "Find all elements according to the given scope and conditions"
+        elements = []
+        ptrs_array = self._element.FindAll(scope, cond)
+        for num in range(ptrs_array.Length):
+            child = ptrs_array.GetElement(num)
+            elements.append(UIAElementInfo(child))
+
+        return elements
+
+    def _build_condition(self, process = None, class_name = None, title = None):
+        "Build UIA filtering conditions"
+        full_cond = _true_condition
+        if process:
+            new_cond = _iuia.CreatePropertyCondition(
+                                    _UIA_dll.UIA_ProcessIdPropertyId, process)
+            full_cond = _iuia.CreateAndCondition(new_cond, full_cond)
             
-        return children
-
-    @property
-    def descendants(self):
-        "Return list of all children for the element"
-        descendants = []
-
-        descendantsArray = self._element.FindAll(_tree_scope['descendants'], _true_condition)
-        for descendantNumber in range(descendantsArray.Length):
-            descendantElement = descendantsArray.GetElement(descendantNumber)
-            descendants.append(UIAElementInfo(descendantElement))
-
-        return descendants
+        if class_name:
+            new_cond = _iuia.CreatePropertyCondition(
+                                    _UIA_dll.UIA_ClassNamePropertyId, class_name)
+            full_cond = _iuia.CreateAndCondition(new_cond, full_cond)
+            
+        if title:
+            # TODO: CreatePropertyConditionEx with PropertyConditionFlags_IgnoreCase
+            new_cond = _iuia.CreatePropertyCondition(
+                                    _UIA_dll.UIA_NamePropertyId, title)
+            full_cond = _iuia.CreateAndCondition(new_cond, full_cond)
+
+        return full_cond
+
+    def children(self, **kwargs):
+        "Return a list of only immediate children of the element according to the criterias"
```

Comment:
typo: "criteria" is already plural. BTW, triple double quotes are preferable by landscape.io for a doc string.